### PR TITLE
bridge and cypress

### PR DESCRIPTION
**Pull request by xithan**
fixes a lot of bridge issues (see PT)
introduces cypress 
tests the bridge with cypress

The cypress installation comes with a lot of example specs. I moved them all to spec/cypress/examples. Although cypress is incredibly well documented I think it's helpful to have them around for a while.

### CODE REVIEW

File: card/mod/account/set/right/account.rb [hunk 1]
```diff
@@ -176,8 +176,8 @@ def token_path_opts
     # Problem: when you do that then the fields are missing in the sign up form:
     # output( [field_nest(:email, view: :titled, title: "email"),
     #          field_nest(:password, view: :titled, title: "password")])
-    %({{+#{:email.cardname}|titled;title:email}}
-      {{+#{:password.cardname}|titled;title:password}})
+    %({{+#{:email.cardname}|edit_row;title:email}}
+      {{+#{:password.cardname}|edit_row;title:password}})
```
Comment (ethn) [inline card/mod/account/set/right/account.rb]: I was hoping to see this grow more idiomatic, but it seems to be growing less.

Is there something really special about  this that justifies all the special handling?

File: card/mod/account/set/right/password.rb [hunk 2]
```diff
@@ -38,8 +38,8 @@ def ok_to_read
 end
 
 format :html do
-  view :core do
-    "<em>#{render_raw}</em"
+  view :core, wrap: :em do
```
Comment (ethn) [inline card/mod/account/set/right/password.rb]: cool.

Comment (ethn) [reply card/mod/account/set/right/password.rb]: If I were to override a view like this, could I get rid of the wrapping?

File: card/mod/account/set/right/account.rb (hunk 1 above)
Comment (ethn) [reply card/mod/account/set/right/account.rb]: I think I'm understanding that this is the edit-in-place mechanism.  But I think we might want edit in place for a number of different views, no?  Eg, we might want something more like `{{+:email|labeled;edit:in-place}}`

(that `+:email` syntax might actually work now...)

File: card/mod/edit/set/all/bridge/tab_views.rb [hunk 3]
```diff
@@ -64,9 +61,14 @@ def bridge_pills items
...
       opts = bridge_link_opts.merge("data-toggle": "pill")
-      add_class opts, "nav-link"
       opts.merge! breadcrumb_data(breadcrumb)
-      opts.merge! extra_opts if extra_opts
+      if extra_opts
+        classes = extra_opts.delete :class
+        add_class opts, classes if classes
+        opts.deep_merge! extra_opts
+      end
+      opts["data-cy"] = "#{text.to_name.key}-pill"
```
Comment (ethn) [inline card/mod/edit/set/all/bridge/tab_views.rb]: might be nice to have all this opts handling in its own method.

File: card/mod/account/set/right/account.rb (hunk 1 above)
Comment (ethn) [reply card/mod/account/set/right/account.rb]: that said, I see how that might not be quick.  seems like we need a consistent way to denote wrapped content so we know what to edit in place.

File: card/mod/machines/lib/javascript/decko_slot.js.coffee [hunk 4]
```diff
@@ -1,3 +1,60 @@
...
+#    _close-overlay
+#    _close-modal
+#
+#   To 3. Similar as 1, the slot has a few overlay and modal options.
+#     classes:
+#     _modal
+#        show slot in modal
+#     _overlay
+#        show slot in overlay
+#
```
Comment (ethn) [inline card/mod/machines/lib/javascript/decko_slot.js.coffee]: good stuff.

File: decko/spec/cypress.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
+  "baseUrl": "http://localhost:5002",
+  "defaultCommandTimeout": 10000,
+  "projectId": "n4h7vq"
+}
```
Comment (ethn) [inline decko/spec/cypress.json]: Can we run this on Semaphore?

File: card/config/initializers/cypress_dev.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+if defined?(CypressDev)
```
Comment (ethn) [inline card/config/initializers/cypress_dev.rb]: Something I need to install?

File: card/mod/account/set/right/account.rb (hunk 1 above)
Comment (xithan) [reply card/mod/account/set/right/account.rb]: Do you have a suggestion how we call the other two values for the edit option?  (the bridge and only the content editor in a modal) Maybe it should be a little bit more self-explanatory than "bridge".

File: card/mod/account/set/right/password.rb (hunk 2 above)
Comment (xithan) [reply card/mod/account/set/right/password.rb]: Aren't these view tags set independent?

File: card/config/initializers/cypress_dev.rb (hunk 6 above)
Comment (xithan) [reply card/config/initializers/cypress_dev.rb]: I don't think so. You only need to know the magic command that will let you enter the testing heaven.

File: card/mod/account/set/right/account.rb (hunk 1 above)
Comment (ethn) [reply card/mod/account/set/right/account.rb]: another minor variant: `content-inline`, `content-modal`, and `full-modal`.

File: card/mod/account/set/right/password.rb (hunk 2 above)
Comment (ethn) [reply card/mod/account/set/right/password.rb]: Looked closer. You extended the `cache` mechanism to handle these, which imo was the right way to go.  That means they're associated with the set and implemented as methods and thus overridable. So I guess all we need is to support `wrap: :none` or something if we want to be able to get rid of wrapping.  (can mark as resolved)

File: decko/spec/cypress.json (hunk 5 above)
Comment (xithan) [reply decko/spec/cypress.json]: yes, we can and we do from now on.

Comment (ethn) [reply decko/spec/cypress.json]: nice!

**Pull request merged**